daphne/src/testing.rs

```diff
@@ -272,3 +303,150 @@ impl DapLeader<BearerToken> for MockAggregator {
         unreachable!("not implemented");
     }
 }
+
+#[derive(Deserialize, Serialize)]
+pub(crate) struct ReportStoreGetPending {
+    pub(crate) reports_requested: u64,
+}
+
+impl From<DapError> for TransitionFailure {
+    fn from(_e: DapError) -> TransitionFailure {
+        TransitionFailure::ReportDropped
+    }
+}
+
+// TODO(nakatsuka-y remove dead_code once all functions are used)
+#[allow(dead_code)]
+pub(crate) struct ReportStore {
+    pending: Arc<Mutex<Vec<Report>>>,
+    processed: Arc<Mutex<HashSet<Nonce>>>,
+    collected: Arc<Mutex<bool>>,
```

Comment:
I'm not sure a mux on each individual field is what we want. For example, it's possible that the leader will add more reports to `pending` a task that has already been collected, which would be an error. How about putting a mux around the entire data structure?

Reply:
You are completely right.
I've moved the mutex to surround `ReportStore`.